mne/tests/test_epochs.py

```diff
@@ -3380,36 +3380,34 @@ def test_empty_constructor():
     BaseEpochs(info, None, None, event_id, tmin, tmax, baseline)
 
 
-def fun(data_epochs):
-    """Auxiliary function to test_apply_function."""
-    nc = np.size(data_epochs, 1)
-    matrix = np.random.rand(nc, nc)
-    return matrix @ data_epochs
-
-
-def fun_ch(data_epochs):
-    """Auxiliary function to test_apply_function channel wise."""
-    sign = -1
-    return data_epochs * sign
-
-
 def test_apply_function():
     """Test apply function to epoch objects."""
-    data = np.random.rand(2, 10, 1000)
+    n_channels = 10
+    data = np.arange(2 * n_channels * 1000).reshape(2, n_channels, 1000)
     events = np.array([[0, 0, 1], [INT32_MAX, 0, 2]])
-    info = mne.create_info(10, 1000., 'eeg')
+    info = mne.create_info(n_channels, 1000., 'eeg')
     epochs = mne.EpochsArray(data, info, events)
     data_epochs = epochs.get_data()
-    # check apply_function in all channels at the time
-    out = epochs.apply_function(fun, channel_wise=False)
-    assert np.shape(out) == np.shape(data_epochs)
-    # check apply_function channel-wise
-    # change sing 3 first channels
-    picks = [0, 1, 2]
-    no_picks = np.arange(3, 10, 1)
-    out_ch = epochs.apply_function(fun_ch, picks=picks, channel_wise=True)
-    out_ch_data = out_ch._data
-    # check whether those channels have been sign inverted
-    check_picks = (np.sign(out_ch_data[:, picks, :]) == -1).all()
-    check_nopicks = (np.sign(out_ch_data[:, no_picks, :]) == 1).all()
-    assert check_picks and check_nopicks
+
+    # apply_function to all channels at once
+    def fun(data):
+        """Reverse channel order without changing values."""
+        return np.eye(data.shape[1])[::-1] @ data
+
+    want = data_epochs[:, ::-1]
+    got = epochs.apply_function(fun, channel_wise=False).get_data()
+    assert np.array_equal(want, got)
+    assert got.shape == data_epochs.shape
```

Comment:
```suggestion
```

Reply:
this is redundant (we already check shape using `_check_fun` when it runs, and `np.array_equal` also checks shape)